core/os/file_access.cpp

```diff
@@ -502,7 +502,8 @@ void FileAccess::store_pascal_string(const String &p_string) {
 
 	CharString cs = p_string.utf8();
 	store_32(cs.length());
-	store_buffer((uint8_t *)&cs[0], cs.length());
+	if (cs.length() > 0)
```

Comment:
Why not `if (p_string.length() == 0) return;` as in `store_string`?

Reply:
From my understanding, pascal string always has the first 4 bytes set to string length. `get_pascal_string` seems to expect that. That's why I don't use the same fix.

Reply:
I think you don't need the condition if changed to:
````store_buffer((uint8_t *)(const char *)cs, cs.length());````
or
````store_buffer((uint8_t *)cs.get_data(), cs.length());````